frame/curve-amm/src/lib.rs

```diff
@@ -157,10 +157,11 @@ pub mod pallet {
 	#[pallet::getter(fn pool_lp_asset)]
 	pub type PoolLPAsset<T: Config> = StorageMap<_, Blake2_128Concat, T::PoolId, T::AssetId>;
 
-	/// List of assets supported by the pool
+	/// Pair of assets supported by the pool
 	#[pallet::storage]
 	#[pallet::getter(fn pool_assets)]
-	pub type PoolAssets<T: Config> = StorageMap<_, Blake2_128Concat, T::PoolId, Vec<T::AssetId>>;
+	pub type PoolAssets<T: Config> =
```

Comment:
how about creating and using pair using struct as in dutch auction?

Reply:
I can update code for DEX if your outstanding PR is merged.

Reply:
@vivekvpandya can you check up on this?